File: gin-vue-admin-main/web/src/api/wl_driver/wlProtocols.js
import service from '@/utils/request'

export const getWlProtocolsList = (params) => {
  return service({
    url: '/wlProtocols/getWlProtocolsList',
    method: 'get',
    params
  })
}

export const createWlProtocols = (data) => {
  return service({
    url: '/wlProtocols/createWlProtocols',
    method: 'post',
    data
  })
}

export const updateWlProtocols = (data) => {
  return service({
    url: '/wlProtocols/updateWlProtocols',
    method: 'put',
    data
  })
}

export const deleteWlProtocols = (params) => {
  return service({
    url: '/wlProtocols/deleteWlProtocols',
    method: 'delete',
    params
  })
}

export const deleteWlProtocolsByIds = (params) => {
  return service({
    url: '/wlProtocols/deleteWlProtocolsByIds',
    method: 'delete',
    params
  })
}

export const findWlProtocols = (params) => {
  return service({
    url: '/wlProtocols/findWlProtocols',
    method: 'get',
    params
  })
} 

File: gin-vue-admin-main/server/router/wl_driver/wl_protocols.go
package wl_driver

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type WlProtocolsRouter struct{}

func (s *WlProtocolsRouter) InitWlProtocolsRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	wlProtocolsRouter := Router.Group("wlProtocols").Use(middleware.OperationRecord())
	wlProtocolsRouterWithoutRecord := Router.Group("wlProtocols")
	{
		wlProtocolsRouter.POST("createWlProtocols", wlProtocolsApi.CreateWlProtocols)
		wlProtocolsRouter.DELETE("deleteWlProtocols", wlProtocolsApi.DeleteWlProtocols)
		wlProtocolsRouter.DELETE("deleteWlProtocolsByIds", wlProtocolsApi.DeleteWlProtocolsByIds)
		wlProtocolsRouter.PUT("updateWlProtocols", wlProtocolsApi.UpdateWlProtocols)
	}
	{
		wlProtocolsRouterWithoutRecord.GET("findWlProtocols", wlProtocolsApi.FindWlProtocols)
		wlProtocolsRouterWithoutRecord.GET("getWlProtocolsList", wlProtocolsApi.GetWlProtocolsList)
	}
}
